Add a few guides to UI Coverage

jennifer-shehane opened the pull request:
Adding guides on common use cases so we can feature them on the intro page.

pstakoun commented on docs/ui-coverage/configuration/attributefilters.mdx:
```diff
@@ -8,13 +8,17 @@ sidebar_label: attributeFilters
+By using attributeFilters, you can ensure UI Coverage selects more appropriate identifiers, leading to cleaner and more accurate coverage reports.
 
-The `attributeFilters` configuration property allows users to specify patterns for attributes and their values that should **not** be used for identifying and grouping elements, thereby allowing UI Coverage to find more suitable identifiers for the impacted elements.
+## Why use attribute filters?
```
attributeFilters?

pstakoun commented on docs/ui-coverage/configuration/attributefilters.mdx:
````diff
@@ -135,9 +117,38 @@ Whether or not a matched attribute should be used for element identification and
+#### HTML
+
+```xml
+<button data-id="852143838" data-cy="user">Harper</button>
````
this is not the best example because data-id is not "significant" by default, so this would not actually do anything.

a better example would be `id` as the dynamic one and ignoring `\\d+` id's, and the buttons also having `name` which is below `id` in priority but would be used if the `id` is filtered out

jennifer-shehane replied:
@pstakoun Can you make an edit to the PR? I'm not quite following every step here in this suggestion.

pstakoun replied:
@jennifer-shehane I pushed an edit to this page

pstakoun commented on docs/ui-coverage/configuration/elementfilters.mdx:
```diff
@@ -8,15 +8,21 @@ sidebar_label: elementFilters
+
 :::info
-Note: setting `elementFilters` impacts both Accessibility and UI Coverage
+**Note:** setting `elementFilters` impacts both Accessibility and UI Coverage
```
impacts both if added to root of config object

pstakoun commented on docs/ui-coverage/configuration/elementfilters.mdx:
```diff
@@ -8,15 +8,21 @@ sidebar_label: elementFilters
-For every element considered interactive and visible by UI Coverage, the first `elementFilters` rule for which the `selector` property matches the element is used to either include or exclude the element based on the `include` value. Elements that do not match any rules are included by default.
+- **Excluding Irrelevant Elements**: Decorative icons, marketing banners, or tooltips that do not require testing can clutter reports. Use elementFilters to exclude them.
+- **Streamlining Reports**: Reducing noise by filtering out non-essential elements makes coverage reports more actionable.
+- **Improving Grouping Accuracy**: Avoid incorrectly grouping elements that share attributes but serve different purposes.
```
not necessarily. another use case is ignoring third party components that you don't need to test

pstakoun replied:
as in "Avoid incorrectly grouping elements that share attributes but serve different purposes" does not sound like something elementFilters helps with

pstakoun commented on docs/ui-coverage/configuration/elementfilters.mdx:
````diff
@@ -161,3 +153,67 @@ A boolean that represents whether or not a matched element should be included in
+
+```xml
+<body>
+  <iframe class="auth908283794"></iframe>
````
this iframe would not be considered interactive anyway, maybe use something with tabindex="0" or a button or something

pstakoun reviewed (commented):
Have not completed review yet, will keep going, but a few things so far

brian-mann commented:
@jennifer-shehane it looks like we're being inconsistent with the "paid addon" across the uicov docs...

Currently the "paid addon" is in the "Introduction", and not in the subsequent docs... 

...but then we use a totally different "paid addon" banner for pages like:

- Results API
- Troubleshooting

**Examples**

![image](https://github.com/user-attachments/assets/8b0d6bc6-d1f8-412b-947e-e90552e8b856)

![image](https://github.com/user-attachments/assets/5e0159df-9fbd-4583-b44a-5f566ac7e696)

jennifer-shehane commented:
@brian-mann Yes, there is not a consistent design decision on how to show this on pages beyond the Get Started pages.

jennifer-shehane commented:
@pstakoun Addressed PR comments in:

- https://github.com/cypress-io/cypress-documentation/pull/6063/commits/838ce1fe790278635f29bd0976521ce34af8e5e2
- https://github.com/cypress-io/cypress-documentation/pull/6063/commits/d788054f06b13a203b6dc961c461a3eea2ce056d

pstakoun commented on docs/ui-coverage/guides/reduce-noise.mdx:
````diff
@@ -0,0 +1,118 @@
+}
+```
+
+To learn more about the configuration options, refer to the [Significant Attributes](/ui-coverage/configuration/significantattributes) documentation.
````
idk which pattern you prefer for naming/referencing the docs. I was using the config property key itself, e.g. significantAttributes, but I see you using a more proper noun name, Significant Attributes. I don't have a strong preference, so long as it is consistent @jennifer-shehane

jennifer-shehane replied:
The display of things in the code ticks can get a little weird, especially if it's in a title. But I don't feel strongly either.

jennifer-shehane commented on docs/ui-coverage/guides/ignore-views-and-links.mdx:
```diff
@@ -0,0 +1,87 @@
+      "pattern": "https://auth.example.com/*",
+      "include": false
+    },
+    {
+      "pattern": "*",
+      "include": true
+    }
```
Yah, I thought this was the case. Some of our config has this end there though so I wasn't sure.

jennifer-shehane commented on docs/ui-coverage/guides/ignore-views-and-links.mdx:
```diff
@@ -0,0 +1,87 @@
+      "pattern": "http://localhost:3000/*#*",
+      "include": false
+    },
+    {
+      "pattern": "http*://*:*/*\\?*#*",
+      "include": false
+    }
```
Does that not exclude all of the CT tests? Since that would match their view paths? I remember needing to add this for the Cypress App for a reason.

pstakoun replied:
viewFilters should not impact CT 🤔

pstakoun replied:
@jennifer-shehane I think this one makes sense to update, since we don't want customers writing `"http*://*:*/*\\?*#*"` in their config

jennifer-shehane commented on docs/ui-coverage/configuration/overview.mdx:
````diff
@@ -67,13 +72,29 @@ You can use the provided editor to write configuration in JSON format. A complet
 ```
 
-Note that the root-level properties `elementFilters` and `viewFilters` apply to both UI Coverage and Accessibility. These properties can also be nested under `uiCoverage` or `accessibility` to apply to only that product, with nested configuration taking precedence over root-level configuration. The `views` property also applies to both products but cannot be nested.
+## Configuration Scope
````
I know, this has ended up being everwhere :/

jennifer-shehane commented on docs/ui-coverage/configuration/viewfilters.mdx:
````diff
@@ -119,3 +114,36 @@ https://cypress.io/login
+#### Config
+
+```json
+{
+  "viewFilters": [
+    {
+      "pattern": "*/404",
+      "include": false
+    },
+    {
+      "pattern": "*/error",
+      "include": false
+    }
+  ]
+}
````
hmmm

jennifer-shehane replied:
Fixed?

jennifer-shehane commented on docs/ui-coverage/configuration/overview.mdx:
```diff
@@ -67,13 +72,29 @@ You can use the provided editor to write configuration in JSON format. A complet
+
+Check out the following configuration guides.
+
+- [Attribute Filters](/ui-coverage/configuration/attributefilters)
+- [Element Filters](/ui-coverage/configuration/elementfilters)
+- [Element Groups](/ui-coverage/configuration/elementgroups)
+- [Elements](/ui-coverage/configuration/elements)
+- [Significant Attributes](/ui-coverage/configuration/significantattributes)
+- [View Filters](/ui-coverage/configuration/viewfilters)
+- [Views](/ui-coverage/configuration/views)
```
Yah, kind of leaning towards the word form. It reads better in some contexts

pstakoun replied:
I'm ok with that

jennifer-shehane commented:
@pstakoun Some fixes addressed: https://github.com/cypress-io/cypress-documentation/pull/6063/commits/133f5f255568ed34acd5d002cb89bbabe3258298

pstakoun commented on docs/ui-coverage/configuration/significantattributes.mdx:
```diff
@@ -8,11 +8,15 @@ sidebar_label: significantAttributes
+## Why use significantAttributes?
 
-The attributes listed in `significantAttributes` are considered "significant" and are prioritized above the default significant attributes, but below `data-cy-ui-id` and `data-cy-ui-group`, for the purpose of identification and grouping.
+## Common use cases
```
should this be ###?

pstakoun commented on docs/ui-coverage/configuration/viewfilters.mdx:
```diff
@@ -8,14 +8,20 @@ sidebar_label: viewFilters
 :::
 
-By default, every URL that is visited within a run is included in UI Coverage. The `viewFilters` property allows you to specify URL patterns for URLs that should be excluded from UI Coverage. Excluding a URL also excludes all links to that URL.
+## Common use cases
```
should this be ###?

pstakoun commented on docs/ui-coverage/configuration/views.mdx:
```diff
@@ -8,17 +8,22 @@ sidebar_label: views
 :::
 
-UI Coverage automatically groups certain URL patterns to create [views](/ui-coverage/core-concepts/views). For URLs that are not automatically grouped (e.g. `/users/alice` and `/users/bob` are not automatically grouped into `/users/*`), the `views` property allows you to specify URL patterns that represent views.
+## Common use cases
```
should this be ###?

pstakoun reviewed (commented):
I don't fully grok the structure of the "Why use X" and then "Common use cases" – there's nothing else under the "Why use X" so do there have to be these double headings?

pstakoun commented on docs/ui-coverage/configuration/elementgroups.mdx:
````diff
@@ -152,3 +150,71 @@ A human-readable name for the group, shown in the elements list.
+
+```xml
+<body>
+  <div id="product125">Product 1</div>
+  <div id="product514">Product 2</div>
+  <div id="product256">Product 3</div>
````
`div` is not interactive. make it `button` or `<div tabindex="0"` or `<div role="button"` or something?

pstakoun commented on docs/ui-coverage/configuration/attributefilters.mdx:
```diff
@@ -8,13 +8,17 @@ sidebar_label: attributeFilters
+By using `attributeFilters`, you can ensure UI Coverage selects more appropriate identifiers, leading to cleaner and more accurate coverage reports.
 
-The `attributeFilters` configuration property allows users to specify patterns for attributes and their values that should **not** be used for identifying and grouping elements, thereby allowing UI Coverage to find more suitable identifiers for the impacted elements.
+## Why use `attributeFilters`?
```
"attribute filters" if doing sentence/word form

pstakoun commented on docs/ui-coverage/configuration/elementgroups.mdx:
```diff
@@ -8,9 +8,16 @@ sidebar_label: elementGroups
+UI Coverage provides logic to automatically [group](/ui-coverage/core-concepts/element-grouping) elements based on their structure in the DOM. However, there are scenarios where you may want to customize these groups to better align with your application's functionality or testing requirements. The `elementGroups` configuration allows you to define custom logic for grouping elements, improving co ...
 
-For every element considered by UI Coverage, the first `elementGroup` rule for which the `selector` property matches the element is used to group the element. Elements that do not match any rules are grouped by the default UI Coverage element grouping rules.
+## Why use `elementGroups`?
```
"element groups" if doing sentence/word form

pstakoun commented on docs/ui-coverage/configuration/significantattributes.mdx:
```diff
@@ -8,11 +8,15 @@ sidebar_label: significantAttributes
+UI Coverage provides logic to automatically [identify](/ui-coverage/core-concepts/element-identification) and [group](/ui-coverage/core-concepts/element-grouping) elements based on their attributes and DOM structure. However, your application may rely on custom attributes for identification that are not prioritized by default in UI Coverage. The `significantAttributes` configuration allows you to ...
 
-Sometimes, your project may use custom attributes to identify elements that are not considered "significant" to the identification and grouping rules used by UI Coverage by default. This can cause UI Coverage to identify and group elements incorrectly, as it does not use the most meaningful attributes available.
+## Why use significantAttributes?
```
"significant attributes" if doing sentence/word form

pstakoun commented on docs/ui-coverage/configuration/viewfilters.mdx:
```diff
@@ -119,3 +114,36 @@ https://cypress.io/login
+      "include": false
+    },
+    {
+      "pattern": "http*://*/error",
```
This should be `http*://*/error/*` ?

pstakoun commented on docs/ui-coverage/configuration/views.mdx:
````diff
@@ -72,29 +68,30 @@ An array of named URL parameters used to further subdivide the URLs into multipl
 
 ```
-https://cypress.io/users/*
+my-app.com/users
+my-app.com/users/*
````
Either

```
https://www.my-app.com/users
https://www.my-app.com/users/*
```

or

```
www.my-app.com/users
www.my-app.com/users/*
```

or

```
/users
/users/*
```

stripping only the `www.` does not make sense / match our logic

pstakoun commented on docs/ui-coverage/configuration/views.mdx:
````diff
@@ -72,29 +68,30 @@ An array of named URL parameters used to further subdivide the URLs into multipl
 #### Config
 
-```json
+```typescript
````
why typescript here?

jennifer-shehane replied:
just a typo

pstakoun commented on docs/ui-coverage/configuration/views.mdx:
```diff
@@ -72,29 +68,30 @@ An array of named URL parameters used to further subdivide the URLs into multipl
 ---
 
-### Using named path parameters
+### Usin named path parameters
```
Mistake?
```suggestion
### Using named path parameters
```

pstakoun commented on docs/ui-coverage/configuration/views.mdx:
````diff
@@ -103,29 +100,30 @@ https://cypress.io/users/*
 ```
-https://cypress.io/users/:name
+my-app.com/users
+my-app.com/users/:name
````
Similar to above, stripping only the `www.` does not make sense

pstakoun commented on docs/ui-coverage/configuration/views.mdx:
````diff
@@ -135,31 +133,31 @@ https://cypress.io/users/:name
 ```
-https://cypress.io/analytics/performance/:id
-https://cypress.io/analytics/usage/:id
+my-app.com/analytics/performance/:id
+my-app.com/analytics/usage/:id
````
same thing re: stripping `www.`

pstakoun commented on docs/ui-coverage/configuration/views.mdx:
````diff
@@ -168,30 +166,30 @@ https://cypress.io/analytics/usage/:id
 ```
-https://cypress.io/app?name=hello
-https://cypress.io/app?name=world
+my-app.com/home?status=done
+my-app.com/home?status=new
````
same thing re: stripping `www.`

pstakoun commented on docs/ui-coverage/core-concepts/views.mdx:
```diff
@@ -9,30 +9,67 @@ sidebar_position: 100
+UI Coverage organizes unique URLs from snapshots in end-to-end tests and mounted components in component tests into Views. Each View represents a distinct page or state in your application, allowing you to track coverage across different parts of your application.
 
-## View Creation & Grouping
+## How views are created
```
We need to decide if "View" is a proper noun or not. It's inconsistent throughout

pstakoun commented on docs/ui-coverage/guides/ignore-elements.mdx:
```diff
@@ -0,0 +1,58 @@
+{
+  "elementFilters": [
+    {
+      "selector": "[data-pendo*]",
```
`data-pendo` is often used for analytics so it actually makes for a good identifier I'd say

jennifer-shehane replied:
Oh, I was interpretting this previous comment as you saying this is a *good* example, now I see you're recommending a change.

pstakoun reviewed (approved):
There are still inconsistencies, but I think it's good enough to merge and iterate since it's blocking other things.
